src/echo-common/src/poe-zone-tracker-service.ts

```diff
@@ -0,0 +1,465 @@
+              map((result) => result.result),
+              map((char) => {
+                if (char != undefined) {
+                  return char
+                } else {
+                  console.log('character found for ' + charName) //Emit this as an event
+                }
+              })
+            )
+            .subscribe((char) => {
+              if (this.currentZoneDelta && char != undefined) {
+                this.currentZoneDelta.characterDelta = {
+                  characterEnter: char,
+                  levelUp: false,
+                  death: false,
+                  inventoryAdded: [],
+                  inventoryRemoved: []
+                }
+
+                if (
+                  this.previousZoneDelta &&
+                  this.previousZoneDelta.characterDelta?.characterEnter
+                ) {
+                  this.previousZoneDelta.characterDelta.characterExit = char
+                  this.setCharacterDelta(
+                    this.previousZoneDelta.characterDelta.characterEnter,
+                    this.previousZoneDelta.characterDelta.characterExit,
+                    this.previousZoneDelta
+                  )
+                  this.zoneExited$.next(this.previousZoneDelta)
+                }
+                this.zoneEntered$.next(this.currentZoneDelta)
+              }
+              this.characterSubscription?.unsubscribe()
+            })
+        } else {
+          if (this.previousZoneDelta) {
+            this.zoneExited$.next(this.previousZoneDelta)
+          }
+          this.zoneEntered$.next(this.currentZoneDelta)
+        }
+      })
+  }
+
+  private setCharacterDelta(enter: PoeCharacter, exit: PoeCharacter, zoneDelta: PoeZoneDelta) {
+    if (!zoneDelta.characterDelta) {
+      throw new Error('Attempt to call setCharacterDelta without delta initialized')
+    }
+
+    if (enter.level && exit.level) {
+      zoneDelta.characterDelta.enterLevel = enter.level
+      zoneDelta.characterDelta.exitLevel = exit.level
+      zoneDelta.characterDelta.levelUp = exit.level > enter.level ? true : false
+    }
+
+    if (exit.experience != undefined && enter.experience != undefined) {
+      zoneDelta.characterDelta.experience = exit.experience - enter.experience
+    }
+
+    const enterMap = itemIdMap(enter.inventory ?? [])
+    const exitMap = itemIdMap(exit.inventory ?? [])
+
+    const added: PoeItem[] = []
+    const removed: PoeItem[] = []
+    enterMap.forEach((item) => {
+      if (item.id && !exitMap.has(item.id)) {
+        removed.push(item)
+      }
+    })
+
+    exitMap.forEach((item) => {
+      if (item.id && !enterMap.has(item.id)) {
+        added.push(item)
+      }
+    })
+
+    zoneDelta.characterDelta.inventoryAdded = added
+    zoneDelta.characterDelta.inventoryRemoved = removed
+  }
+
+  private npcEncounterEventSubscriber() {
+    this.logEvents$
+      .pipe(
+        filter((e) => e.type == 'NPCEncounterEvent'),
+        map((e) => e as PoeNPCEncounterEvent)
```

Comment:
Checks like these could be converted to `if(!e.type === "XXXXX") return;` in the subscribe, since the filter/map aren't doing anything special in the pipe.

Reply:
Seems like a better way to handle it in general yes